core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -311,6 +379,26 @@ public synchronized InetSocketAddress getAddress() throws UnavailableException {
     return mAddress;
   }
 
+  /**
+   * Get the server address for connection.
+   * <p>
+   * Note: Some client does not use {@link InetSocketAddress} for connection, e.g.,
+   * BlockWorkerClient might connect through a domain socket. This is the place where
+   * inheriting classes overrides the logic of getting whatever address they need to connect
+   * to.
+   * <p>
+   * @return the {@link GrpcServerAddress} for connection
+   * @throws UnavailableException when no address can be determined
+   */
+  protected synchronized GrpcServerAddress getGrpcServerAddress() throws UnavailableException {
+    mAddress = getAddress();
+    if (mAddress != null) {
+      return GrpcServerAddress.create(mAddress);
+    } else {
+      throw new UnavailableException("Cannot Determine Server Address");
+    }
+  }
```

Comment:
As commented, not all clients talk to a remote service through internet, this method exposes a way for inheriting classes to define their own address to connect to.